nixos/modules/services/web-apps/lasuite-docs.nix

```diff
@@ -0,0 +1,473 @@
+        ''
+    }
+    ${optionalString (
+      cfg.s3.accessKeyIDPath != null
+    ) "export AWS_S3_ACCESS_KEY_ID=$(cat ${cfg.s3.accessKeyIDPath})"}
+    ${optionalString (
+      cfg.s3.secretAccessKeyPath != null
+    ) "export AWS_S3_SECRET_ACCESS_KEY=$(cat ${cfg.s3.secretAccessKeyPath})"}
+    ${optionalString (
+      cfg.oidc.clientSecretPath != null
+    ) "export OIDC_RP_CLIENT_SECRET=$(cat ${cfg.oidc.clientSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.serverSecretPath != null
+    ) "export COLLABORATION_SERVER_SECRET=$(cat ${cfg.collaborationServer.serverSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.yproviderApiKeyPath != null
+    ) "export Y_PROVIDER_API_KEY=$(cat ${cfg.collaborationServer.yproviderApiKeyPath})"}
+  '';
+in
+{
+  options.services.lasuite-docs = {
+    enable = mkEnableOption "SuiteNumérique Docs";
+
+    package = mkPackageOption pkgs "lasuite-docs" { };
+
+    frontendPackage = mkPackageOption pkgs "lsauite-docs-frontend" { };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        Address used by gunicorn to listen to.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8000;
+      description = ''
+        Port used by gunicorn to listen to.
+      '';
+    };
+
+    enableNginx = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable Nginx as a proxy server.
+      '';
+    };
+
+    secretKeyPath = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path to the Django secret key.
+      '';
+    };
+
+    s3 = {
+      url = mkOption {
+        type = types.str;
+        description = ''
+          URL of the s3 bucket.
+        '';
+      };
+
+      accessKeyIDPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the access key ID of the bucket.
+        '';
+      };
+
+      secretAccessKeyPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the secret access key of the bucket.
+        '';
+      };
+    };
+
+    oidc = {
+      clientSecretPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the client secret of the client of OIDC.
+        '';
+      };
+    };
+
+    database = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Configure local PostgreSQL database server for docs.
+        '';
+      };
+    };
+
+    redis = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Configure local Redis cache server for docs.
+        '';
+      };
+    };
```

Comment:
This should mention that the connection is set up in other options, as the redis namespace seems confusing otherwise.